File: fitness-frontend/src/types/foodLibrary.ts
import type { FoodCategory } from './enums'

/**
 * 食物热量库
 * <p>
 * 库由 `sql/init.sql` 播种 30 条；饮食记录只能引用库中已存在的食物名（精确匹配）。
 */
export interface FoodItem {
  id: number
  foodName: string
  category: FoodCategory | string
  /** 每 100g 热量(kcal) */
  caloriesPer100g: number
  proteinPer100g: number | null
  fatPer100g: number | null
  carbsPer100g: number | null
  /** yyyy-MM-dd HH:mm:ss */
  createdAt: string
  updatedAt: string
}

/** 食物库查询参数（keyword 与 category 可同时生效） */
export interface FoodLibraryQuery {
  /** 食物名模糊匹配 */
  keyword?: string
  category?: string
}

/** 食物库列表响应 */
export interface FoodLibraryResponse {
  list: FoodItem[]
  /** 全库分类（后端已按固定顺序 主食/肉类/蔬菜/水果/乳制品/零食/饮品 排好） */
  categories: string[]
}

/** 重建缓存响应（返回实际加载条数，失败时接口会返回错误码而不是假成功） */
export interface ReloadCacheResponse {
  loaded: number
}

File: src/main/java/com/fitness/entity/FoodLibrary.java
package com.fitness.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 食物热量库实体 — 对应 t_food_library 表
 */
@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "t_food_library")
public class FoodLibrary {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /** 食物名称 */
    @Column(nullable = false, length = 100, unique = true)
    private String foodName;

    /** 分类 */
    @Column(nullable = false, length = 20)
    private String category;

    /**
     * 每100g热量(kcal)
     * <p>
     * ⚠️ <b>必须显式写 name</b>：隐式物理名推导（Spring Boot 默认的
     * {@code CamelCaseToUnderscoresNamingStrategy}）只在「小写→大写→小写」处插下划线，
     * 数字不构成词边界，因此 {@code caloriesPer100g} 会被推导成 {@code calories_per100g}，
     * 而 {@code sql/init.sql} 里的列名是 {@code calories_per_100g}。
     * 由于 {@code ddl-auto: none}（DDL 由 init.sql 手工管理），Hibernate 不会校验，
     * 不一致的表现是运行期 MySQL 1054 {@code Unknown column}，
     * 且测试用的 H2 是 create-drop（由实体反推建表），根本测不出来。
     */
    @Column(name = "calories_per_100g", nullable = false, precision = 6, scale = 1)
    private BigDecimal caloriesPer100g;

    /** 每100g蛋白质(g) — 同上，隐式名会得到 {@code protein_per100g}，故显式指定 */
    @Column(name = "protein_per_100g", precision = 5, scale = 1)
    private BigDecimal proteinPer100g;

    /** 每100g脂肪(g) — 同上 */
    @Column(name = "fat_per_100g", precision = 5, scale = 1)
    private BigDecimal fatPer100g;

    /** 每100g碳水(g) — 同上 */
    @Column(name = "carbs_per_100g", precision = 5, scale = 1)
    private BigDecimal carbsPer100g;

    @Column(updatable = false, columnDefinition = "DATETIME DEFAULT CURRENT_TIMESTAMP")
    private LocalDateTime createdAt;

    @Column(columnDefinition = "DATETIME DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP")
    private LocalDateTime updatedAt;
}
